sprint_15

YanikMax opened the pull request.

Commit:
Перенесен код из старого проекта

Commit:
1) Пересоздал проект, сделал первый коммит и перенес в новый файл код до исправления.
2) Кнопки + и datePicker перенес в NavBarItem
3) Добавил возможность скрывать клавиатуру в строке поиска и в строке названия трекера
4) Добавил реализацию скролла таблицы с расписанием на маленьких экранах
5) Добавлен лейбл счетчика, а так же склонение слова "дня"
6) Добавил реализацию запрета отметки карточки трекера для будущих дат
7) Строка поиска теперь работает (реализовано нажатием на Fix)
8) Убраны force unwrapping из кода

Данила, прошу прощения, что некорректно создал репозиторий. Сейчас это новый репозиторий с исправлением тех ошибок, что Вы указали.

Commit:
1) Внес исправления в метод configureNavigationBar()
2) Удалил настройки для datePicker и addButton для view и из списка на добавление на view
3) Обернул TrackersViewController в UINavigationController на этапе добавления в иерархию
P.S. Данила, если у меня снова будет проблема с NavBar напишите, пожалуйста, в комментариях на GitHub, мол, ты дебил и ничего не исправил, чтобы мне не договариваться о еще 1 попытке сдачи. Спасибо.

Commit:
1) Завершены работы по экрану "новая привычка". Добавлены эмоджи, цвета и экран "Расписание".
2) Добавлен слой для работы с БД

LizaIlinykh commented on CoreData/TrackerRecordStore.swift:
```diff
@@ -0,0 +1,79 @@
+    // MARK: - Methods
+    func add(_ newRecord: TrackerRecord) throws {
+        let trackerCoreData = try trackerStore.getTrackerCoreData(by: newRecord.trackerId)
+        let TrackerRecordCoreData = TrackerRecordCoreData(context: context)
```
именования экземпляров должны быть с маленькой буквы

LizaIlinykh commented on CoreData/TrackerStore.swift:
```diff
@@ -0,0 +1,175 @@
+            id: id,
+            label: label,
+            emoji: emoji,
+            color: color!,
```
рекомендую избегать принудительных операций

LizaIlinykh commented on TrackerYP/Tracker/TrackerViewController.swift:
```diff
@@ -0,0 +1,352 @@
+    }
+    
+    @objc private func didChangedDatePicker(_ sender: UIDatePicker) {
+        currentDate = Date.from(date: sender.date)!
```
рекомендую избегать принудительных операций

Commit:
1) Исправлено именование константы.
2) Избавился от принудительного извлечения в TrackerStore, TrackerViewController.
3) При выборе дней в расписании теперь они отображаются на экране создания привычки.
